### ISSUE TITLE
Document is null from v3.0.0

### ISSUE THREAD

**Issue by phmasek** [sanity-io/sanity-plugin-iframe-pane#89]
**Describe the bug**

After updating to `>v3.0.0` the document property for the url option is null.

**To Reproduce**

```typescript
// defaultDocumentNode.ts
// ...

return S.document().views([
  S.view.form(),
  S.view
    .component(Iframe)
    .options({
      url: (document: SanityDocument  /*  <-- This document is null */) => resolveProductionUrl(document, context),
    })
    .title('Preview'),
]);
```

**Expected behavior**

Document should not be null.

**Which versions of Sanity are you using?**

```terminal
@sanity/cli (global)                   3.20.1 (up to date)
@sanity/document-internationalization   2.0.2 (up to date)
@sanity/eslint-config-studio            3.0.1 (up to date)
@sanity/vision                         3.20.1 (up to date)
sanity                                 3.20.1 (up to date)
```

**Which versions of Node.js / npm are you running?**

```terminal
9.6.7
v18.17.1
```

**Additional context**

Works fine with version `2.6.1`.

**Comment by salex97** [comment]
I've encountered the same issue but only when the document is published and no draft exists. When a draft exists my getPreviewUrl works like it should.

**Comment by spacedawwwg** [comment]
Looking at the source code, it seems to be built around 'draft' rather than 'displayed' on the SanityDocument

**Comment by rdunk** [comment]
Reported as still not fixed in #91

@phmasek  @salex97 Are you able to confirm if `3.8.2` fixes the issue for you?

**Comment by phmasek** [comment]
@rdunk looks like it's working for me at `v3.1.2`.
